lib/react_client.dart

```diff
@@ -583,25 +615,130 @@ ReactDartComponentFactoryProxy _registerComponent(ComponentFactory componentFact
   return new ReactDartComponentFactoryProxy(reactComponentClass);
 }
 
+class ReactJsContextComponentFactoryProxy extends ReactJsComponentFactoryProxy {
+  /// The JS class used by this factory.
+  @override
+  final ReactClass type;
+  final bool isConsumer;
+  final bool isProvider;
+  final Function factory;
+  final bool convertDomProps;
+
+  ReactJsContextComponentFactoryProxy(
+    ReactClass jsClass, {
+    this.convertDomProps: true,
+    this.isConsumer: false,
+    this.isProvider: false,
+  })  : this.type = jsClass,
+        this.factory = React.createFactory(jsClass),
+        super(jsClass, convertDomProps: convertDomProps);
+
+  @override
+  ReactElement build(Map props, [List childrenArgs]) {
+    dynamic children = _convertArgsToChildren(childrenArgs);
+
+    if (isConsumer) {
+      if (children is Function) {
+        Function contextCallback = children;
+        children = allowInterop((args) {
+          return contextCallback(_unjsifyNewContext(args));
+        });
+      }
+    }
+
+    return factory(generateExtendedJsProps(new Map.from(props)), children);
+  }
+
+  /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
+  /// consumption by the [react] library internals.
+  generateExtendedJsProps(Map props) {
+    JsBackedMap propsForJs = new JsBackedMap.from(props);
+
+    if (isProvider) {
+      propsForJs['value'] = _jsifyNewContext(propsForJs['value']);
+    }
+
+    if (convertDomProps) {
```

Comment:
Is it valid to be able to pass DOM props to a context component? If not, can we remove this? Same question below with `ref`.